src/scriptworker/context.py

```diff
@@ -237,17 +240,22 @@ def event_loop(self, event_loop: asyncio.AbstractEventLoop) -> None:
     async def populate_projects(self, force: bool = False) -> None:
         """Download the ``projects.yml`` file and populate ``self.projects``.
 
-        This only sets it once, unless ``force`` is set.
+        Unless ``force`` is set or it was last fetched over an hour ago this
+        method is a noop. (The latter is needed to avoid using a stale version
+        of this file on long lived workers forever.)
 
         Args:
             force (bool, optional): Re-run the download, even if ``self.projects``
                 is already defined. Defaults to False.
 
         """
         assert self.config
-        if force or not self.projects:
+        now = time.time()
+        last_fetched = now - self._projects_timestamp
+        if force or not self.projects or last_fetched > 3600:
```

Comment:
nit: I would make the `3600` here a const up top so it's easier to spot/adjust if necessary.